### PR TITLE
Feature/update pom

### PR DESCRIPTION
**Pull request by jweiser**
Reorganize and streamline the POM file. All details are in the commit messages.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -120,8 +128,34 @@
...
 	<build>
 		<plugins>
+			<!-- sets the Java compiler version targeted by Maven -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-compiler-plugin</artifactId>
+				<version>3.8.1</version>
+				<configuration>
+					<source>1.8</source>
+					<target>1.8</target>
```
Comment (SolomonShorser-OICR) [inline pom.xml:154]: Oh, I _just_ realized you're doing this _everywhere_ now...

File: pom.xml [hunk 2]
```diff
@@ -173,6 +213,18 @@
 					<autoReleaseAfterClose>false</autoReleaseAfterClose>
 				</configuration>
 			</plugin>
+
+			<!-- executes unit tests -->
+			<plugin>
```
Comment (SolomonShorser-OICR) [inline pom.xml:218]: If this isn't getting any special configuration, why explicitly list it?

File: pom.xml (hunk 1 above)
Comment (jweiser) [reply pom.xml:154]: Yeah, I removed the maven compiler properties in favour of stating it explicitly in the compiler plugin in the all the projects.

Comment (SolomonShorser-OICR) [reply pom.xml:154]: Yeah, I thought you were only tweaking one project.

File: pom.xml (hunk 2 above)
Comment (jweiser) [reply pom.xml:218]: It was required to have the tests run when executing `mvn test` or any lifecycle phase that calls it.  Without the surefire plugin stated explicitly, the tests didn't run when I executed `mvn test`.

Comment (SolomonShorser-OICR) [reply pom.xml:218]: Oh, really? I thought tests were supposed to run by default unless the `-DskipTests` option was given. Was this required for unit tests or integration tests? Now that I think about it... I think unit tests are supposed to run by default, and integration tests are not. Or maybe they run by default if the phase is _not_ `test`?

Comment (SolomonShorser-OICR) [reply pom.xml:218]: Or... maybe the default behaviour has changed...

Comment (jweiser) [reply pom.xml:218]: I thought so, too.  The tests were run by default for some of the other projects but not others like this one.  I'm not sure why, but explicit surefire configuration was able to get them working.  I may come back to take a closer look into what's going on, though.

For this project, all the tests are unit tests.  Running both `mvn test` and `mvn clean package` without the surefire plugin included produced output that no tests were run but maven still ran to completion successfully.

Comment (SolomonShorser-OICR) [reply pom.xml:218]: Really? Weeeeeeird. Well, if adding the plugin works, go for it! :)

Comment (jweiser) [reply pom.xml:218]: > Or... maybe the default behaviour has changed...

I did see something online about changes to the maven surefire plugin, but I think something else may be going on that is specific to the projects that didn't run their tests by default.

Comment (jweiser) [reply pom.xml:218]: > Really? Weeeeeeird. Well, if adding the plugin works, go for it! :)

Yeah, I would like to figure out why.  It makes me uneasy that it didn't work or not work consistently without the maven surefire plugin specified explicitly.

File: pom.xml (hunk 1 above)
Comment (jweiser) [reply pom.xml:154]: I just thought it would be good to be consistent since I'm going over all the projects anyways :).

**Review by cookersjs — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (SolomonShorser-OICR) [reply pom.xml:154]: Yeah... I like the other style. But, whatever... :man_shrugging: ;)

**Review by SolomonShorser-OICR — APPROVED** [review]

**Pull request merged**